packages/cli/src/services/redis/redis-client.service.ts

```diff
@@ -23,7 +22,6 @@ export class RedisClientService {
 		return client;
 	}
 
-	@OnShutdown()
```

Comment:
Could we instead of removing this, set a `priority = 0`?

Reply:
The current setup prevents us from registering `disconnectClients` with `ShutdownService`

```ts
await Promise.all([this.stopProcess(), this.shutdownService.waitForShutdown()]);
```

Even if we assign priority `0` so that `RedisClientService` is the last to shut down inside `waitForShutdown`, that is still no guarantee that `stopProcess` will always reach `queue.pause` and the graceful shutdown before `disconnectClients` is called.

Reply:
Discussed internally, going with a third option: [4dc60af](https://github.com/n8n-io/n8n/pull/9928/commits/4dc60af3079a4923828a4af571e855b2332b35ad)